src/components/ReviewComponents/ReviewComponentsPage.tsx

```diff
@@ -1,22 +1,51 @@
 import * as React from 'react';
 import { useHistory } from 'react-router-dom';
-import { Formik } from 'formik';
-import { useFormValues } from '../form-context';
-import { useWizardContext } from '../Wizard/Wizard';
+import { Formik, FormikProps } from 'formik';
+import { mapDetectedComponents, useComponentDetection } from '../AddComponent/utils';
+import { containerImageRegex } from '../AddComponent/validation-utils';
+import { DetectedComponentData, useFormValues } from '../form-context';
 import { ReviewComponentsForm } from './ReviewComponentsForm';
 import { createResources } from './submit-utils';
-import { ReviewComponentsFormValues } from './types';
+import { ComponentValues, ReviewComponentsFormValues } from './types';
 import { transformComponentValues } from './utils';
 import { reviewFormSchema } from './validation-utils';
 
 export const ReviewComponentsPage: React.FC = () => {
-  const { decreaseStepBy } = useWizardContext();
   const [formState] = useFormValues();
-  const isSample = formState.components[0].type === 'sample';
   const history = useHistory();
 
+  const isContainerImage = containerImageRegex.test(formState.source);
+
+  const [detectedComponents, loadError] = useComponentDetection(
+    !isContainerImage ? formState.source : null,
+    formState.application,
+    formState.git,
+  );
+
+  const [components, loaded]: [DetectedComponentData[], boolean] = React.useMemo(() => {
+    if (isContainerImage) {
+      const sourceLength = formState.source.split('/').length;
+      return [
+        [
+          {
+            data: { source: { image: { containerImage: formState.source } } },
+            name: formState.source.split('/')?.[sourceLength - 1],
+          },
+        ] as DetectedComponentData[],
+        true,
+      ];
+    }
+    if (detectedComponents && !loadError) {
+      return [mapDetectedComponents(detectedComponents), true];
+    }
+    if (loadError) {
+      return [[], true];
```

Comment:
Should we show some indication that no components were detected?
In the previous flow we would have the source field in an error state, but here it would show empty page.

Reply:
Yes, That sounds good, we will need UX input on this.